common/djangoapps/student/forms.py

```diff
@@ -188,6 +188,19 @@ def __init__(
                                 "required": _("To enroll, you must follow the honor code.")
                             }
                         )
+                elif field_name == 'data_sharing_consent':
+                    if field_value == "required":
+                        self.fields[field_name] = TrueField(
+                            error_messages={
+                                "required": _(
+                                    "You must consent to data sharing to register."
```

Comment:
To be clear, this message shouldn't show up in regular use; just when we attempt to create users via the API. without UI-side validation. I'm open to changing it to whatever is suggested.